src/storage/storage.cc

```diff
@@ -64,18 +64,20 @@ Storage::Handle StorageImpl::Alloc(size_t size, Context ctx) {
   std::shared_ptr<storage::StorageManager> manager = device.Get(
       ctx.dev_id, [ctx]() {
         storage::StorageManager *ptr = nullptr;
+#if MXNET_USE_CUDA
+        num_gpu_device = 0;
+        cudaError_t e = cudaGetDeviceCount(&num_gpu_device);
```

Comment:
Ok, then we would need to duplicate that code for both GPU and CPUPinned paths. I assume this is ok?